src/lib/cache/redis/redis.go

```diff
@@ -107,23 +91,45 @@ func New(opts cache.Options) (cache.Cache, error) {
 		opts.Address = "redis://localhost:6379/0"
 	}
 
-	name := fmt.Sprintf("%x", sha256.Sum256([]byte(opts.Address)))
-
-	param := &libredis.PoolParam{
-		PoolMaxIdle:           100,
-		PoolMaxActive:         1000,
-		PoolIdleTimeout:       10 * time.Minute,
-		DialConnectionTimeout: time.Second,
-		DialReadTimeout:       time.Second * 2,
-		DialWriteTimeout:      time.Second * 5,
+	u, err := url.Parse(opts.Address)
+	if err != nil {
+		return nil, err
 	}
 
-	pool, err := libredis.GetRedisPool(name, opts.Address, param)
+	// For compatibility, should convert idle_timeout_seconds to idle_timeout.
+	values, err := url.ParseQuery(u.RawQuery)
 	if err != nil {
 		return nil, err
 	}
 
-	return &Cache{opts: &opts, pool: pool}, nil
+	if t := values.Get("idle_timeout_seconds"); t != "" {
+		values.Del("idle_timeout_seconds")
+		values.Set("idle_timeout", t)
+		u.RawQuery = values.Encode()
+	}
+
+	var client *redis.Client
+
+	switch u.Scheme {
```

Comment:
since newer version redis supports username & password in the url, not sure whether the error raised here contains the sensitive message.

Reply:
the parse function will never print any username/password related fields.